app/controllers/france_connect/particulier_controller.rb

```diff
@@ -14,112 +15,136 @@ def login
 
-        flash.notice = t('france_connect.particulier.flash.connection_done', application_name: Current.application_name)
-        connect_france_connect_particulier(user)
-      end
-    else
-      flash.alert = t('france_connect.particulier.flash.invalid_password')
-    end
+    render :confirmation_sent, locals: { email: @fci.email_france_connect, destination_path: destination_path(@fci.user) }
   end
 
-  def mail_merge_with_existing_account
-    user = User.find_by(email: sanitize(@fci.email_france_connect.downcase))
-    if user.can_france_connect?
-      @fci.update(user: user)
-      @fci.delete_merge_token!
+  def merge_using_password
+    user = User.find_by(email: sanitize(@fci.email_france_connect))
+
+    if user.present? && !user.can_france_connect?
+      return destroy_fci_and_redirect_to_login(@fci)
+    end
+
+    if user.present? && user.valid_for_authentication? { user.valid_password?(params[:password]) }
+      @fci.safely_update_user(user:)
 
       flash.notice = t('france_connect.particulier.flash.connection_done', application_name: Current.application_name)
       connect_france_connect_particulier(user)
-    else # same behaviour as redirect nicely with message when instructeur/administrateur
-      @fci.destroy
-      redirect_to new_user_session_path, alert: t('errors.messages.france_connect.forbidden_html', reset_link: new_user_password_path)
+    else
+      flash.alert = t('france_connect.particulier.flash.invalid_password')
     end
   end
 
-  def merge_with_new_account
-    user = User.find_by(email: sanitized_email_params)
+  def merge_using_email_link
+    user = User.find_by(email: @fci.requested_email)
 
-    if user.nil?
-      @fci.associate_user!(sanitized_email_params)
-      @fci.delete_merge_token!
+    if user.present? && !user.can_france_connect?
+      return destroy_fci_and_redirect_to_login(@fci)
+    end
 
-      flash.notice = t('france_connect.particulier.flash.connection_done', application_name: Current.application_name)
-      connect_france_connect_particulier(@fci.user)
+    if user.nil?
+      @fci.safely_associate_user!(@fci.requested_email)
     else
-      @email = sanitized_email_params
-      @merge_token = merge_token_params
+      @fci.safely_update_user(user:)
     end
+
+    @fci.user.update(email_verified_at: Time.zone.now)
+
+    flash.notice = t('france_connect.particulier.flash.connection_done', application_name: Current.application_name)
+    connect_france_connect_particulier(@fci.user)
   end
 
-  def resend_and_renew_merge_confirmation
-    @fci.create_email_merge_token!
-    UserMailer.france_connect_merge_confirmation(
-      @fci.email_france_connect,
-      @fci.email_merge_token,
-      @fci.email_merge_token_created_at
-    )
-      .deliver_later
+  # TODO mutualiser avec le controller Users::ActivateController
+  # pour toute la partie de confirmation de compte
+  def confirm_email
+    if @user.confirmation_sent_at && 2.days.ago < @user.confirmation_sent_at
```

Comment:
```suggestion
    if @user.confirmation_sent_at && @user.confirmation_sent_at > 2.days.ago
```

C'est peut-être que moi, mais l'autre forme, je n'arrive pas à passer dans ma tête

Reply:
C'est marrant, je viens de remarquer qu'il y a pas mal d'endroits dans le code où c'est fait avec la date en premier. Donc, c'est vraiment moi qui doit avoir un blocage dans la tête. Vous pouvez ignorer.